internal/appsec/waf.go

```diff
@@ -103,16 +104,39 @@ func newHTTPWAFEventListener(handle *waf.Handle, addresses []string, timeout tim
 
 	return httpsec.OnHandlerOperationStart(func(op *httpsec.Operation, args httpsec.HandlerOperationArgs) {
 		var body interface{}
+		values := map[string]interface{}{}
 		wafCtx := waf.NewContext(handle)
 		if wafCtx == nil {
 			// The WAF event listener got concurrently released
 			return
 		}
 
-		values := map[string]interface{}{}
+		// OnUserIDOperationStart happens when appsec.SetUser() is called. We run the WAF and apply actions to
+		// see if the associated user should be blocked. Since we don't control the execution flow in this case
+		// (SetUser is SDK), we delegate the responsibility of interrupting the handler to the user.
+		op.On(sharedsec.OnUserIDOperationStart(func(operation *sharedsec.UserIDOperation, args sharedsec.UserIDOperationArgs) {
+			values := map[string]interface{}{}
+			for _, addr := range addresses {
```

Comment:
Bonus: see about transforming addresses to a map

Reply:
This won't be included in this change - adding a ticket to our backlog since this would be nice to have.